pennylane/templates/tensornetworks/mps.py

```diff
@@ -0,0 +1,179 @@
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the MPS template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import warnings
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+
+
+def compute_indices_MPS(wires, loc):
+    r"""
+    Generate a list containing the wires for each block.
+
+    Args:
+        wires (Iterable): wires that the template acts on
+        loc (int): number of wires per block
+    Returns:
+        layers (array): array of wire indices or wire labels for each block
+    """
+
+    layers = np.array(
+        [
+            [wires[idx] for idx in range(j, j + loc)]
+            for j in range(0, len(wires) - int(len(wires) % (loc // 2)) - loc // 2, loc // 2)
+        ]
+    )
+    return layers
+
+
+class MPS(Operation):
+    r"""Quantum circuit that broadcasts local gates, similar to the architecture in `arXiv:1803.11537 <https://arxiv.org/abs/1803.11537>`_.
+
+    The argument ``block`` is a user-defined quantum function. ``block`` must include two arguments: ``block_weights`` and ``block_wires``.
+
+    Args:
+        wires (Iterable): wires that the template acts on
+        loc (int): number of wires per block
+        block (Callable): quantum circuit that composes a block
+        n_params_block (int): the number of parameters in a block; equal to the number of elements in ``block_weights``
+        weights (Sequence): list containing the weights for all blocks; weights should have one element per block
+
+    .. Usage Details::
+
+        This example demonstrates the use of ``MPS`` for a simple block.
+
+        .. code-block:: python
+
+            import pennylane as qml
+            import numpy as np
+
+            def block(block_weights, block_wires):
+                qml.CNOT(wires=[block_wires[0],block_wires[1]])
+                qml.Rot(block_weights[0],block_weights[1],block_weights[2],wires=block_wires[0])
+                qml.Rot(block_weights[3],block_weights[4],block_weights[5],wires=block_wires[1])
+
+            n_wires = 4
+            loc = 2
+            n_params_block = 6
+            template_weights = [[1,2,3,4,5,6],[3,4,5,6,7,8],[4,5,6,7,8,9]]
+
+            dev= qml.device('default.qubit',wires=n_wires)
+            @qml.qnode(dev)
+            def circuit(weights):
+                qml.MPS(wires = range(n_wires),loc=loc,block=block, n_params_block=n_params_block, weights=weights)
```

Comment:
We can just use `qml.MPS(wires = range(n_wires),loc, block, n_params_block, weights)`, right?